odbc/src/api/functions.rs

```diff
@@ -3573,21 +3672,21 @@ unsafe fn sql_set_stmt_attrw_helper(
     let stmt = must_be_valid!(stmt_handle.as_statement());
     match attribute {
         StatementAttribute::SQL_ATTR_APP_ROW_DESC => {
-            stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_APP_ROW_DESC"));
+            add_diag_with_function!(stmt_handle,ODBCError::Unimplemented("SQL_ATTR_APP_ROW_DESC"), "SQLSetStmtAttrW");
             SqlReturn::ERROR
         }
         StatementAttribute::SQL_ATTR_APP_PARAM_DESC => {
-            stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_APP_PARAM_DESC"));
+            add_diag_with_function!(stmt_handle,ODBCError::Unimplemented("SQL_ATTR_APP_PARAM_DESC"), "SQLSetStmtAttrW");
             SqlReturn::ERROR
         }
         StatementAttribute::SQL_ATTR_IMP_ROW_DESC => {
             // TODO: SQL_681, determine the correct SQL state
-            stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_IMP_ROW_DESC"));
+            add_diag_with_function!(stmt_handle,ODBCError::Unimplemented("SQL_ATTR_IMP_ROW_DESC"), "SQLSetStmtAttrW");
             SqlReturn::ERROR
         }
         StatementAttribute::SQL_ATTR_IMP_PARAM_DESC => {
             // TODO: SQL_681, determine the correct SQL state
-            stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_IMP_PARAM_DESC"));
+            add_diag_with_function!(stmt_handle,ODBCError::Unimplemented("SQL_ATTR_IMP_PARAM_DESC"), "SQLSetStmtAttrW");
             SqlReturn::ERROR
         }
         StatementAttribute::SQL_ATTR_CURSOR_SCROLLABLE => {
```

Comment:
This function is missing a few places where we need to replace `stmt_handle.add_diag_info()` with `add_diag_with_function!()`.  A few spots here and some more closer to the end.